### DISCUSSION TITLE
Replacement of excessive use of string parameters

### DISCUSSION THREAD

**Discussion by thomasdorone** [godotengine/godot-proposals#5571] [Scripting]
Hi,

I absolutely love the engine so far, it's straightforward, clean, and clear. The only thing that annoys me is the excessive use of string parameters in functions. I think that most of them could easily be replaced either by callbacks or directly accessing the properties.

For example why instead of:
`this.Set("parameters/MyBlend/blend_amount",1f);`

Can't we simply write something like:
`GetNode<AnimationNodeBlend2>("Myblend").blend_amount=1f;`

And why can't signals be callbacks?

**Comment by Calinou** [comment]
> And why can't signals be callbacks?

I believe this has been addressed in 4.0.beta, but this can't be backported to 3.x as it uses Mono instead of .NET 6.
